cli/cmd/project/refresh.go

```diff
@@ -113,6 +116,11 @@ func RefreshCmd(ch *cmdutil.Helper) *cobra.Command {
 				return fmt.Errorf("the --full flag can only be used with --all or --model")
 			}
 
+			// Return an error for ineffective use of --partition
+			if !cmd.Flags().Changed("model") {
+				return fmt.Errorf("--partition requires --model to be explicitly set")
+			}
```

Comment:
This breaks the command for all other resource types than a model (e.g. `rill project refresh --source NAME`)! Also I don't think this is even needed, see my earlier comment here: https://github.com/rilldata/rill/pull/6657#discussion_r1977997711